src/core/infer_request.h

```diff
@@ -304,7 +309,12 @@ class InferenceRequest {
   void SetTrace(std::unique_ptr<InferenceTrace>&& trace)
   {
     trace_ = std::move(trace);
+#ifdef TRITON_ENABLE_TRACING
+    response_factory_.SetTrace(std::move(trace_->CopyTrace()));
```

Comment:
If we need to have the trace object be used by both request and response(s), then we should change the trace to be a shared pointer everywhere. We should not be creating copies of the trace object. The potential complication is that the activity collection and other functions may need additional synchronization as the response(s) may be reporting trace activities/tensors interleaved with each other and with the request.